src/components/ec-letter-icon/ec-letter-icon.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <component
+    :is="isClickable ? 'button' : 'div'"
+    data-test="ec-letter-icon"
+    class="ec-letter-icon"
+    :class="{ 'ec-letter-icon--clickable': isClickable }"
+    :style="containerSize"
+  >
+    <span
+      data-test="ec-letter-icon__text"
+      class="ec-letter-icon__text"
+      :style="letterSize"
+    >
+      {{ firstLetter }}
+    </span>
+  </component>
+</template>
+
+<script setup lang="ts">
+import { computed, type StyleValue } from 'vue';
+
+import type { LetterIconProps } from './types';
+
+const props = withDefaults(defineProps<LetterIconProps>(), {
+  size: 24,
+  isClickable: false,
+});
+
+const firstLetter = computed(() => (props.text[0] ? props.text[0].toUpperCase() : ''));
```

Comment:
```suggestion
const firstLetter = computed(() => (props.text[0]?.toUpperCase() || ''));
```

Reply:
Why do we need to cover the "empty string" scenario? Can't we throw if an empty string or do not render it at all?

Reply:
Mainly to avoid an error on Storybook as it is possible to set an empty string through the controls

Reply:
We also could hide the icon at all if the string is empty, I agree.